In [1]:
import pandas as pd

airports = pd.read_csv("airports.csv")
weather = pd.read_csv("weather.csv")

airports.head()

faa                           name        lat        lon   alt  tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264  -6   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801  -6   A   
3  06N                Randall Airport  41.431912 -74.391561   523  -5   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11  -5   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York

In [2]:
airports.columns

Index(['faa', 'name', 'lat', 'lon', 'alt', 'tz', 'dst', 'tzone'], dtype='object')

In [3]:
northernmost = airports.sort_values(by="lat", ascending=False)

northernmost[["faa", "name", "lat"]].head(5)

faa                                        name        lat
417  EEN                     Dillant Hopkins Airport  72.270833
230  BRW                  Wiley Post Will Rogers Mem  71.285446
110  AIN                          Wainwright Airport  70.638056
708  K03                               Wainwright As  70.613378
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300

In [4]:

# Remove impossible latitude values (continental US does not reach 70+ except Alaska)

clean_airports = airports[airports["lat"] < 72]

clean_northernmost = clean_airports.sort_values(by="lat", ascending=False)

clean_northernmost[["faa", "name", "lat"]].head(5)

faa                                        name        lat
230   BRW                  Wiley Post Will Rogers Mem  71.285446
110   AIN                          Wainwright Airport  70.638056
708   K03                               Wainwright As  70.613378
152   ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300
1363  UUK                        Ugnu-Kuparuk Airport  70.330833

##1. What is the northernmost airport in the United States?
  After sorting the airports dataset by latitude in descending order, the airport with the highest latitude initially appeared to be Dillant Hopkins Airport with a latitude of 72.27°N. However, after verifying this airport’s location, I found that it is actually located in New Hampshire at approximately 42.9°N. This indicates that the latitude value in the dataset is incorrect and represents a data entry error.
  Because this value is unrealistic and geographically impossible, I treated it as an outlier and removed it from consideration. After excluding this and sorting again, the airport with the highest valid latitude is Wiley Post Will Rogers Memorial Airport in Utqiagvik, Alaska, with a latitude of 71.285°N.
  More research confirms that this airport is located in northern Alaska and is recognized as the northernmost public airport in the United States.
  Therefore, the northernmost airport in the dataset (after correcting for data errors) is Wiley Post Will Rogers Memorial Airport.


In [5]:
easternmost = airports.sort_values(by="lon", ascending=False)

easternmost[["faa", "name", "lon"]].head(5)

faa                           name         lon
1290  SYA                   Eareckson As  174.113620
942   MYF               Montgomery Field  117.759000
396   DVT  Deer Valley Municipal Airport  112.457000
417   EEN        Dillant Hopkins Airport   42.898333
444   EPM     Eastport Municipal Airport  -67.012694

In [6]:
# Remove airports with positive longitude values
western_hemisphere = airports[airports["lon"] < 0]

easternmost = western_hemisphere.sort_values(by="lon", ascending=False)

easternmost[["faa", "name", "lon"]].head(5)

faa                                 name        lon
444   EPM           Eastport Municipal Airport -67.012694
624   HUL                         Houlton Intl -67.792056
259   CAR                         Caribou Muni -68.017917
1101  PQI  Northern Maine Rgnl At Presque Isle -68.044797
1398  WFK  Northern Aroostook Regional Airport -68.312778

##2. What is the easternmost airport in the United States?

  When I first sorted the dataset by longitude in descending order, the airport that appeared first was Eareckson Air Station with a longitude of 174.11°. However, this value is positive because the Aleutian Islands cross the 180° meridian. Although the airport is located in Alaska, its coordinate system makes it appear extremely far east numerically.
  Since I’m suppose to be interpreting  “easternmost” using standard western hemisphere longitude values, I excluded airports with positive longitude values and focused only on airports with negative longitude (west of the Prime Meridian).
After filtering and sorting again, the airport with the highest valid longitude value was Eastport Municipal Airport (EPM) in Eastport, Maine, with a longitude of approximately −67.01°.
  More research confirmed that Eastport, Maine is the easternmost city in the contiguous United States. Therefore, the easternmost airport in the dataset (after accounting for longitude interpretation) is Eastport Municipal Airport.

In [7]:
feb12 = weather[
    (weather["year"] == 2013) &
    (weather["month"] == 2) &
    (weather["day"] == 12)
]

feb12.head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1006    EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1007    EWR  2013      2   12     1  39.92  37.94  92.56     250.0   
1008    EWR  2013      2   12     2  39.92  28.04  62.21     270.0   
1009    EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
1010    EWR  2013      2   12     4  39.02  26.96  64.29     280.0   

      wind_speed  wind_gust  precip  pressure  visib             time_hour  
1006     6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1007     9.20624        NaN     0.0    1007.2   10.0  2013-02-12T06:00:00Z  
1008    20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z  
1009  1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
1010    12.65858        NaN     0.0    1008.8   10.0  2013-02-12T09:00:00Z

In [8]:
feb12["wind_speed"].describe()

count      72.000000
mean       28.577703
std       121.947200
min         4.603120
25%        12.370885
50%        14.960140
75%        17.261700
max      1048.360580
Name: wind_speed, dtype: float64

In [9]:
# Remove unrealistic wind speeds
feb12_clean = feb12[feb12["wind_speed"] < 200]

feb12_clean["wind_speed"].describe()

count    71.000000
mean     14.214564
std       4.217477
min       4.603120
25%      12.083190
50%      14.960140
75%      17.261700
max      23.015600
Name: wind_speed, dtype: float64

In [10]:
feb12_clean.groupby("origin")["wind_speed"].max()

origin
EWR    21.86482
JFK    20.71404
LGA    23.01560
Name: wind_speed, dtype: float64

## 3. On February 12th, 2013, which New York area airport had the windiest weather?
  First, I filtered the weather dataset for February 12, 2013. When I checked the summary statistics for wind_speed, I noticed the maximum value was 1048 mph, which is clearly impossible and must be a data error.
  Since that value is not realistic, I treated it as an outlier and removed wind speeds greater than 200 mph. After cleaning the data, the maximum recorded wind speed was about 23 mph, which is reasonable for winter weather in New York.
  Next, I grouped the cleaned data by airport and calculated the maximum wind speed for each location. The results showed EWR: 21.86 mph, JFK: 20.71 mph, and LGA: 23.02 mph. Based on all of this, LGA had the windiest weather on February 12, 2013.